### ISSUE TITLE
Inlay hints should display inferred type argument

### ISSUE THREAD

**Issue by Zekfad** [Dart-Code/Dart-Code#4366]
**Is your feature request related to a problem? Please describe.**
It's hard to navigate to inferred type of generics. You have to lookup relevant argument and find type from it.
For now it's only possible to see inferred type from hover popup, but it's just plain text.

**Describe the solution you'd like**
Inlay hints should show clickable inferred types of generics.

**Describe alternatives you've considered**
Make hovers interactive? (But that's likely out of scope).

**Additional context**

<details>
<summary>Screenshot of current behavior</summary>

![image](https://user-images.githubusercontent.com/8970959/215269630-189bb1e4-350b-4c3e-bd1a-2ed48b851866.png)

</details>

**Comment by DanTup** [comment]
Do you mean in cases like this?

```dart
foo<T>(T a) {
  foo('');         // Show an inlay hint of `<String>` after `foo` so this reads foo<String>('')
}
```

**Comment by Zekfad** [comment]
Hi @DanTup, thanks for fast reply, yes, that's what I meant.

**Comment by DanTup** [comment]
I've a change open to add this for type arguments and also list/map/set literals where it was missing before.

![Feb-21-2023 12-36-15](https://user-images.githubusercontent.com/1078012/220346634-e27ce3d2-f054-435d-b909-9486eb2eedd7.gif)

https://dart-review.googlesource.com/c/sdk/+/284143

**Comment by Zekfad** [comment]
Hi @DanTup, thank you for your work!
Is there ETA for Preview version with this feature?

**Comment by DanTup** [comment]
The change is in the Dart analysis server in the SDK, so once the review is done and the code landed, it'll be available in [main channel](https://dart.dev/get-dart/archive#main-channel) builds of the SDK usually within a few hours or so (and Flutter's `master` branch probably within a few days). I wouldn't recommend using those versions for production work but they're good for testing. Of course the change will also then show up in future dev/beta/stable releases of the SDKs.

**Comment by DanTup** [comment]
Landed in https://github.com/dart-lang/sdk/commit/e8a8e10bcdcca75d1b3a7daf3e7a40d885b51e64.

**Comment by Zekfad** [comment]
Hi @DanTup, on your screenshot, it infers <int>a: [1,2], shouldn't it be a: <int>[1,2]?

**Comment by DanTup** [comment]
@Zekfad Oof, yes, I hadn't noticed that (and apparently I didn't write tests when there are both type args and parameter names at the same offset). I will fix, thanks!

**Comment by DanTup** [comment]
Got a fix open at https://dart-review.googlesource.com/c/sdk/+/284641.

**Comment by DanTup** [comment]
Fix landed in https://github.com/dart-lang/sdk/commit/658e34ee2acaa88074f589dbbadf647904e719d3.
